stl/src/vector_algorithms.cpp

```diff
@@ -78,6 +81,106 @@ namespace {
 
 extern "C" {
 
+#ifdef _M_ARM64
+__declspec(noalias) void __cdecl __std_swap_ranges_trivially_swappable_noalias(
+    void* _First1, void* const _Last1, void* _First2) noexcept {
+    constexpr size_t _Mask_64 = ~((static_cast<size_t>(1) << 6) - 1);
+    if (_Byte_length(_First1, _Last1) >= 64) {
+        const void* _Stop_at = _First1;
+        _Advance_bytes(_Stop_at, _Byte_length(_First1, _Last1) & _Mask_64);
+        do {
+            const uint8x16_t _Left1  = vld1q_u8(static_cast<uint8_t*>(_First1) + 0);
+            const uint8x16_t _Left2  = vld1q_u8(static_cast<uint8_t*>(_First1) + 16);
+            const uint8x16_t _Left3  = vld1q_u8(static_cast<uint8_t*>(_First1) + 32);
+            const uint8x16_t _Left4  = vld1q_u8(static_cast<uint8_t*>(_First1) + 48);
+            const uint8x16_t _Right1 = vld1q_u8(static_cast<uint8_t*>(_First2) + 0);
+            const uint8x16_t _Right2 = vld1q_u8(static_cast<uint8_t*>(_First2) + 16);
+            const uint8x16_t _Right3 = vld1q_u8(static_cast<uint8_t*>(_First2) + 32);
+            const uint8x16_t _Right4 = vld1q_u8(static_cast<uint8_t*>(_First2) + 48);
+            vst1q_u8(static_cast<uint8_t*>(_First1) + 0, _Right1);
+            vst1q_u8(static_cast<uint8_t*>(_First1) + 16, _Right2);
+            vst1q_u8(static_cast<uint8_t*>(_First1) + 32, _Right3);
+            vst1q_u8(static_cast<uint8_t*>(_First1) + 48, _Right4);
+            vst1q_u8(static_cast<uint8_t*>(_First2) + 0, _Left1);
+            vst1q_u8(static_cast<uint8_t*>(_First2) + 16, _Left2);
+            vst1q_u8(static_cast<uint8_t*>(_First2) + 32, _Left3);
+            vst1q_u8(static_cast<uint8_t*>(_First2) + 48, _Left4);
+            _Advance_bytes(_First1, 64);
+            _Advance_bytes(_First2, 64);
+        } while (_First1 != _Stop_at);
+    }
+
+    constexpr size_t _Mask_32 = ~((static_cast<size_t>(1) << 5) - 1);
+    if (_Byte_length(_First1, _Last1) >= 32) {
+        const void* _Stop_at = _First1;
+        _Advance_bytes(_Stop_at, _Byte_length(_First1, _Last1) & _Mask_32);
+        do {

```

Comment:
I see a minor efficiency improvement here. This structure is following the x64/x86 code (which is great; our codebase greatly values consistency which makes reviewing and avoiding bugs so much easier). However, that code has to deal with AVX2 and SSE4.2 potentially not being available, so each step might need to loop.

For ARM64, the structure is much more uniform. Only the initial 64-byte step needs to loop. After that, we have at most 63 bytes remaining, so if we can consume a 32-byte chunk, there will be only one. After that, we have at most 31 bytes remaining, so if we can consume a 16-byte chunk, there will be only one, etc.

This will allow us to avoid some branches, which should be a pure perf win.